### ISSUE TITLE
🤑 Run Fund Fest 2023

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/team-ospo#108]
[← 2022](https://github.com/getsentry/team-ospo/issues/37) | [Notion🔒](https://www.notion.so/sentry/Funding-Open-Source-5ebce9bcb55849ca8e7d9ff61c32a12b) | [Sheet🔒](https://docs.google.com/spreadsheets/d/1efxcQPg_4bG-zWh59m_-1Bz7cawLD36oJcLEPV9-Sa8/edit?usp=sharing)・[Sheet 2🔒](https://docs.google.com/spreadsheets/d/1ek_Tz1nnGR1pSkTVasDtPcN5eaVP-vcJQlzH26rM8AA/edit?usp=sharing) | [2024 →](https://github.com/getsentry/team-ospo/issues/163)

![](https://github.com/getsentry/team-ospo/assets/134455/1a57c7a2-dd78-4435-bc72-44e12112b76c)

## Greetings!

Welcome to Sentry's annual Open Source Fund Fest for 2023! This program takes a lot of work to run and this issue is the central source of truth for coordinating all of the moving parts. This year our budget is **$500,000**, which is almost double [last year](https://blog.sentry.io/we-just-gave-260-028-dollars-to-open-source-maintainers/). There are two key components to this year's edition:

1. **[GitHub Sponsors](https://github.com/sponsors)**—We are continuing our partnership with GitHub, and this year our goal is to become **the first company to approach 100% coverage** of all of our dependencies on Sponsors, across three orgs (`gesentry`, `codecov`, and `syntaxfm`). We are dedicating 10% of our budget to this, so these will be small amounts but lots of them and will show as coming directly from us. We expect a number of these to be the [first sponsorship that people ever receive](https://twitter.com/fux0r/status/1709841482507509992). <br><br><img width="300"  alt="Screenshot 2023-10-06 at 10 21 50 AM" src="https://github.com/getsentry/team-ospo/assets/134455/644f1cb7-bb0e-41c3-9f86-12b6fcbb7415"><img width="300" alt="Screenshot 2023-10-06 at 10 21 14 AM" src="https://github.com/getsentry/team-ospo/assets/134455/636ba9b1-34a0-4463-b638-c27badd42756"><br><br>

1. **[Thanks.dev](https://thanks.dev/)**—We were a launch partner for this new platform last year, and our pilot project went so well that we have **contracted with Thanks.dev to manage the bulk of our budget** this year. The size and scope of our program is quite significant, and the product Thanks.dev is building and the level of expertise they are developing in this field make them an excellent partner for us. We're inventing the future together!<br><img width="400" alt="ecosystem boost" src="https://github.com/getsentry/team-ospo/assets/134455/b745234c-dc0a-438f-b45a-2bfb3580b58b"><br><br>

Thanks.dev offers payouts through Stripe (0% additional fee) and Open Collective (10%) and is adding payouts through GitHub Sponsors for us (targeting the 3% fee tier). You will need to [sign up on Thanks.dev](https://thanks.dev/home) if you want to receive more than the small amounts we're sending to everyone directly through GitHub Sponsors.

We're aiming to publish our wrap-up blog post on October 24. In the mean time, feel free to comment on this issue or [reach out on 𝕏](https://twitter.com/chadwhitacre_) with questions. **Thank you** to all of the maintainers who build the software Sentry depends on. 🙏 

### FOSS Funders

While it is fun and exciting to run arguably the most comprehensive and robust program in the industry for really truly funding Open Source, the only way we're going to solve Open Source sustainability once and for all is for _every_ company to step up and participate. Sentry is part of the [FOSS Funders](https://fossfunders.com/) working group to drive this change. If your company has an Open Source funding story to tell, [please join us](https://github.com/fossfunders/fossfunders.com#readme)!

## To Do

- [x] July 15 - Confirm [budget](https://twitter.com/chadwhitacre_/status/1682030221757911041)
- [x] July 15 - Update [AP calendar🔒](https://www.notion.so/sentry/Funding-Open-Source-5ebce9bcb55849ca8e7d9ff61c32a12b?pvs=4#a54d8f386e044bc3b382c0763759d45a), determine invoice due dates for below.
- [x] August 1 - Kick off internal survey, run through August 18 (close before hackweek)
- [x] Receive and pay invoices
    - [x] August 11 - zeroth batch due, missed it this year, hit next year?
    - [x] August 25 - first batch due, including first from GitHub
    - [x] September 8 - second batch due, including **final** from GitHub
    - [x] September 22 - third batch due - **aim for all by here** - now just one, from Thanks.dev! 😁
    - [x] October 6 - last chance, ~try hard not to have to use it!~ - second one from TD, we decided to split into two since they're large, make sure they make it through the banking system okay
- [x] Local meetups that Sentry employees frequent? - handled in dev survey
- [x] Review "[What's new with GitHub Sponsors](https://github.blog/2023-04-04-whats-new-with-github-sponsors/)"—[bulk sponsorships](https://docs.github.com/en/sponsors/sponsoring-open-source-contributors/sponsoring-an-open-source-contributor#sponsoring-accounts-in-bulk)!
- [x] platforms?
    - [x] GHS
    - [x] ~OC~
    - [x] ~stackaid~
    - [x] thanks.dev!
    - [x] ~polar.sh~—not ready, maybe next year
- [ ] foundations
    - [ ] pay
        - [ ] Python
        - [x] JavaScript
        - [x] Rust
        - [x] PostgreSQL
        - [x] Apache
        - [x] PHP
        - [x] Ruby
        - [x] OSI
        - [x] Outreachy
    - [ ] logo - https://sentry.io/branding/
        - [ ] Python
        - [ ] JavaScript
        - [ ] Rust
        - [x] ~PostgreSQL~
        - [ ] Apache
        - [ ] PHP
        - [ ] Ruby
        - [ ] OSI
        - [ ] Outreachy
- [ ] GHS
    - [x] sort out [sponsor linking](https://github.com/orgs/getsentry/sponsoring/settings) (GH Enterprise feature?)
    - [x] create sponsorships for `getsentry`
    - [x] get balances transferred to `codecov` and `syntaxfm`
    - [x] create sponsorships for `codecov`
    - [x] create sponsorships for `syntaxfm`
    - [ ] double-check final totals
- [x] TD
    - [x] Don't forget (these are things Sentry employees or others have specifically mentioned to me over the past year) ...
        - [x] `codecov` and `syntaxfm` orgs
        - [x] last year's recipients - [e.g.](https://twitter.com/ljharb/status/1710862163844505639)
        - [x] [Fastlane/MNF](https://github.com/MobileNativeFoundation/discussions/discussions/194)
        - [x] [Moka](https://github.com/moka-rs/moka)
        - [x] rrweb - vital for Replay, but we may fund them through contracting? Find out details and record for posterity
        - [x] https://github.com/sponsors/webknjaz ([X](https://twitter.com/webknjaz/status/1710259402840674812))
        - [x] https://opencollective.com/json-schema ([X](https://twitter.com/relequestual/status/1710006059148914903))
        - [x] [ICCF Holland](https://iccf-holland.org/index.html)? ([because](https://groups.google.com/g/vim_announce/c/tWahca9zkt4)) - not sure they're recruitable, made a personal donation
    - [x] review draft weighting
- [x] [publish blog post🔒](https://getsentry.atlassian.net/servicedesk/customer/portal/17/DTSD-276)
    - [x] [update hero🔒](https://getsentry.atlassian.net/servicedesk/customer/portal/17/DTSD-295) - https://brand.getsentry.com/share/M7C2Nwq1Jypfe1tcT7QC

### GitHub Sponsorships

1. "[A'ight, beat that. 😤😁💃](https://twitter.com/chadwhitacre_/status/1710378308012023998)" (chadwhitacre_)
2. "[I'm sorry, I don't make the rules. 🐭](https://twitter.com/chadwhitacre_/status/1710399485073846493)" (chadwhitacre_)
3. "[Sentry is giving $500k+ to open source it relies on this year](https://twitter.com/zeeg/status/1710764429519049192)" (zeeg)
4. "[it’s insane that people are anything other than thankful](https://twitter.com/jarredsumner/status/1711159005698240616)" (jarredsumner)
5. "[A company donates $500K/year in direct funding to OSS](https://twitter.com/gergelyorosz/status/1711271491071848936)" (gergelyorosz)
6. "[YOU ARE A PIECE OF SHIT FOR NOT DONATING MORE OF YOUR VC FUNDING](https://twitter.com/ThePrimeagen/status/1711458245691285568)" (ThePrimeagen)

### Announcement

1. "[We Just Gave $500,000 to Open Source Maintainers](https://blog.sentry.io/we-just-gave-500-000-dollars-to-open-source-maintainers/)"
1. [X](https://twitter.com/getsentry/status/1716860343237116014)
2. [HN](https://news.ycombinator.com/item?id=38001924) - 112 points and 36 comments despite being [brutally modded](https://twitter.com/chadwhitacre_/status/1716874002751500441) at ~20 points - and then [modded a second time](https://twitter.com/chadwhitacre_/status/1716901027394245025)! [rankings](https://hnrankings.info/38001924/)
3. [/r/opensource/](https://www.reddit.com/r/opensource/comments/17fmsry/we_just_gave_500000_to_open_source_maintainers/)
4. "[It's the 2nd year in a row we've hit the front page only to have HN mods remove it.](https://twitter.com/bentlegen/status/1716884717394670015)" (bentlegen)
5. "[The answer I got last year (and kudos dang for the tone)](https://twitter.com/chadwhitacre_/status/1716947994338021885)" (chadwhitacre_)
6. "[Tarsnap has given 2^18 dollars to open source](https://www.daemonology.net/blog/2023-10-25-2%5E18-dollars-to-open-source.html)" (cperciva) - [HN](https://news.ycombinator.com/item?id=38010244) (not modded! 😁 390 / 180 / [rankings](https://hnrankings.info/38010244/)) - [X](https://twitter.com/cperciva/status/1717082393390268519)

**Comment by chadwhitacre** [comment]
Alright, slogging through the thick fog of confusion, hopefully emerging?

I've decided not to use StackAid this year, not seeing enough momentum and I need to focus my efforts. 🐭

I really like what I see at Thanks.dev. It's basically a productization of the spreadsheet I've been using and the team has been super-responsive, basically building it to spec for me. ❤️ 🙏

GitHub Sponsors is making gradual progress but is still quite cumbersome for a program at our scale, no-where near as useful as Thanks.dev.

OpenCollective is approximately as useful as GHS and the fees are higher, 10% vs. 3%.

Two years ago I wrote [deps](https://github.com/getsentry/deps) and did allocations manually. Last year I reran deps but also put a little through TD and SA as pilots. I didn't try to reconcile recipients and amounts between them, I just let the TD/SA money be free bonus extra cash dollar dough bucks. This year I would like to flip it:

1. **~12k** — Achieve ≥ 99% coverage on GHS for `getsentry`, `codecov`, and `syntaxfm`, but with a limited dollar amount, basically just buying logo placement.
1. **~120k** — Do some direct payments to foundations still, though a couple of these can be run through GHS or OC.
1. **~368k** — Run the rest through Thanks.dev so we get nice weighting.

Needs from TD:

1. Mix in survey results.
2. Passthrough to OC/GHS where the recipient is not on TD (I guess with extra 3% or 10% haircut).
3. Weighting by repo topic ([`tag-production`]())
4. Think about slew?

Needs for TD:

1. Security approval to install Thanks.dev on private repos (nice to have).
2. Install Thanks.dev on all public repos.

For GitHub:

1. Write a script to use the GraphQL API to create sponsorships.

**Comment by chadwhitacre** [comment]
I have $50k locked up in GHS right now, I should be able to use for PSF and DSF, not sure I'll be able to transfer to other orgs though.

**Comment by chadwhitacre** [comment]
Engaging with Thanks.dev. Sent in email:

> 1. Thanks.dev (TD) administers Sentry's Open Source funding program for the year.
> 2. $25k fee
> 3. $475k to be allocated and disbursed via one-time or monthly recurring payouts
> 4. $50k is already located in GitHub Sponsors (GHS), remainder ($425k) will be transferred to TD, OC, GHS, or direct to project as necessary
> 5. Recipients determined via automated dependency analysis in three GitHub orgs + additional lists (survey results, SDKs, prior year).
> 6. Allocations weighted according to Sentry's priorities (orgs, repos, ecosystems, projects, recipients)
> 7. Payout mechanisms include TD (0% additional fee), OC (10%), GitHub Sponsors (3%), and direct payments (0%) to a limited number of projects (~10).
> 8. All funds disbursed (for one-time payouts) or configured (for monthly payouts) by Oct 10.
> 
> Let me know if we need to align further on details. Do you have an order form that we can start with? If not [we] can provide. I am seeing [terms](https://thanks.dev/terms) on your site, not sure if we can start there or what, will let [legal] drive from here.

Internal procurement ticket: https://getsentry.atlassian.net/servicedesk/customer/portal/15/PSD-294

**Comment by chadwhitacre** [comment]
@nehzata Can we use this ticket to coordinate the rest of our work together this year?

I believe the contract is in your hands to docusign. Once that is inked we are ready to run!

The [the thanks-dev app](https://github.com/apps/thanks-dev/) is installed on `getsentry`, `codecov`, and `syntaxfm`. However, the @getsentry-funding account is not a member of the latter two orgs. I have a request in to @jeffrey-sentry to help out with that. I also seem to be having some trouble getting the account through our new SCIM workflow on the `getsentry` org as well, so I think we may be fully blocked on this.

I had a call with GitHub about transfering funds between the three orgs, I added a sheet to the [spreadsheet](https://docs.google.com/spreadsheets/d/1efxcQPg_4bG-zWh59m_-1Bz7cawLD36oJcLEPV9-Sa8/edit?usp=sharing) ("GHS") with details, tl;dr targeting $4/maintainer/month across all three orgs. Need to investigate minimums in our set of maintainers and decide on our approach (easiest is to drop any maintainer with a minimum > $4/mo, but we have some flex if needed).

I also added contact emails for the foundations on the "Summary" tab. Feel free to start reaching out to folks and let me know if any of the target amounts do not line up with levels/expectations for this year. For Outreachy they work in 6-month intervals so the 20k will need to be 2 x 10k I think.

**Comment by nehzata** [comment]
Thanks @chadwhitacre! Contract is signed now. Exciting!

On our side we've nearly finished the manual inclusions process to support survey + sdk deps. We'll get the GH sponsors script sorted next and @anehzat will be starting with the foundations.

**Comment by chadwhitacre** [comment]
Sounds good. I am traveling today and all next week but I have my laptop, we're so async with timezones in the best of times that I doubt you'll notice much anyway, but in particular I will watch for email threads with foundations that @anehzat starts and chime in as needed. Thanks, @nehzata @anehzat! Good luck! 😁 ☘️ 🙏

**Comment by chadwhitacre** [comment]
lol should've checked my email first ... love you guys! 😁

<img width="725" alt="Screenshot 2023-09-15 at 4 08 35 PM" src="https://github.com/getsentry/team-ospo/assets/134455/2fc523e6-810f-4a70-9067-f0c3dc53031a">

**Comment by nehzata** [comment]
Hi @jeffrey-sentry! Hope you're well. 

Following @chadwhitacre's previous message re the app installation (https://github.com/getsentry/team-ospo/issues/108#issuecomment-1720095976), we can confirm access to `syntaxfm` on our side but not `getsentry` & `codecov`.

Please let me know if there's anything we can do on our side to assist. 🙏

**Comment by jeffrey-sentry** [comment]
@nehzata I do not have access to the `codecov` org, but I just checked the thanks.dev app permission setting for `getsentry` and it does have `Read` access to all repos. Is there anything else we need to do on our side beyond that?

Thanks!

**Comment by chadwhitacre** [comment]
Here are the repos I see when I login to thanks.dev using @getsentry-funding. I guess we need to do the dance to see `codecov` and `getsentry` there.

<img width="703" alt="Screenshot 2023-09-19 at 7 19 01 PM" src="https://github.com/getsentry/team-ospo/assets/134455/f81ca4a6-fc9e-40db-8457-e80cf743c473">

**Comment by chadwhitacre** [comment]
Here's what I see when I try to configure the app using @getsentry-funding from the thanks.dev UI:

---------

<img width="711" alt="Screenshot 2023-09-19 at 7 32 02 PM" src="https://github.com/getsentry/team-ospo/assets/134455/2fc387ca-f6ef-4c9a-8e0b-9bdc203c2a79">

---------

<img width="597" alt="Screenshot 2023-09-19 at 7 32 08 PM" src="https://github.com/getsentry/team-ospo/assets/134455/c1424b53-a9ac-4746-b391-a43f197979a8">

---------

<img width="588" alt="Screenshot 2023-09-19 at 7 31 07 PM" src="https://github.com/getsentry/team-ospo/assets/134455/72f462a7-73e0-451a-abb5-08f4d2ad14d9">

**Comment by chadwhitacre** [comment]
When I configure using my account I see that it is installed on all repos (for all three orgs).

<img width="855" alt="Screenshot 2023-09-19 at 7 33 06 PM" src="https://github.com/getsentry/team-ospo/assets/134455/a100cf61-3755-4e19-80e9-36bfc020e765">

**Comment by chadwhitacre** [comment]
@nehzata Was there a manual step last year on your side to get this wired up? I forget but have a vague recollection. Any thoughts on how to proceed?

**Comment by nehzata** [comment]
@chadwhitacre looks like for the Sentry org it expects us to manually select the repos individually. Last year we had a script running that would populate everything without requiring the app installation. That worked as all the repos we were interested in were public. If I understood correctly, this year there are private repos as well? If not shall we revert to the previous process?

**Comment by nehzata** [comment]
Just to clarify it seems we have the following options atm:
1. Use last year's process
2. Switch to gh/chadwhitacre profile
3. Manually configure the repos for Sentry (by clicking on Sentry > Configure > "Only select repositories" and adding the relevant repos)
4. Wait for access?

**Comment by nehzata** [comment]
> @nehzata I do not have access to the `codecov` org, but I just checked the thanks.dev app permission setting for `getsentry` and it does have `Read` access to all repos. Is there anything else we need to do on our side beyond that?
> 
> Thanks!

Thanks for checking it out @jeffrey-sentry!

**Comment by nehzata** [comment]
@chadwhitacre We submitted a bug report with GitHub regarding the issacs entry and got the following response.

![image](https://github.com/getsentry/team-ospo/assets/72539235/f8c4bc31-5b2b-4bf4-a23c-7916da5d5238)

**Comment by nehzata** [comment]
@chadwhitacre First version of GH mass sponsorship app is ready to test [here](https://github.com/thnxdev/wkr-gh-sponsor)! :)

**Comment by chadwhitacre** [comment]
> We submitted a bug report with GitHub regarding the issacs entry and got the following response.

What org am I a collaborator in that results in isaacs showing up in the list? I don't see `isaacs` on the list of orgs on https://github.com/apps/thanks-dev/installations/select_target, e.g. Feel free to cc me on the support ticket and I can connect with GH support directly.

> First version of GH mass sponsorship app is ready to test [here](https://github.com/thnxdev/wkr-gh-sponsor)! :)

Dope! Looks like it's designed to run continuously, yes? Can I run it once to set up recurring monthly donations?

> for the Sentry org it expects us to manually select the repos individually

Is it some config specific to the `getsentry` org that requires this? How about `codecov`? I have the app set to "all repos" in both orgs (as well as `syntaxfm`) so I am wondering where specifically we are tripped up here.

**Comment by nehzata** [comment]
> What org am I a collaborator in that results in isaacs showing up in the list? I don't see `isaacs` on the list of orgs on https://github.com/apps/thanks-dev/installations/select_target, e.g. Feel free to cc me on the support ticket and I can connect with GH support directly.

The original ticket is closed as it's from Nov last year. I did some further digging into this and looks like the following is happening:
- You're a collaborator on `github.com/isaacs/github`;
- isaacs has previously logged into TD and has therefore installed the app on his account;
- GH sees your account as a collaborator of isaacs' RepositoryOwner entity;
- Therefore your installation token also by default gets access to all the repos in that entity that you have read access on via the GraphQL API (which are all his public repos);

We've come across many such disconnects between token scopes in the App installation API & GraphQL API unfortunately...

```
curl -s -H 'authorization: bearer <TOKEN>' https://api.github.com/repos/isaacs/github/collaborators | jq '. | map(.login)'
[
  "schacon",
  "wfarr",
  "clarkbw",
  "holman",
  "isaacs",
  "haacked",
  "aspiers",
  **"chadwhitacre"**,
  "tjfontaine",
  "TPS",
  "dreww",
  "broccolini",
  "joernhees",
  "jlord",
  "cirosantilli",
  "yoannchaudet",
  "michellemerrill"
]
```

> > First version of GH mass sponsorship app is ready to test [here](https://github.com/thnxdev/wkr-gh-sponsor)! :)
> 
> Dope! Looks like it's designed to run continuously, yes? Can I run it once to set up recurring monthly donations?

Yes it currently runs continuously to cater for network failures + GH access token rate limiting. Recurring donations [may](https://docs.github.com/en/graphql/reference/input-objects#createsponsorshipinput:~:text=Whether%20the%20sponsorship%20should%20happen%20monthly/yearly%20or%20just%20this%20one%20time.%20Required%20if%20a%20tierId%20is%20not%20specified.) be possible but we don't know if it works yet as the corresponding monthly/yearly setting are not present in the API. We've [asked](https://github.com/orgs/community/discussions/67767) in GH community forums. Should be a one or two line change.

Let's do a quick code review + test to confirm once you're back online please?

> > for the Sentry org it expects us to manually select the repos individually
> 
> Is it some config specific to the `getsentry` org that requires this? How about `codecov`? I have the app set to "all repos" in both orgs (as well as `syntaxfm`) so I am wondering where specifically we are tripped up here.

Not really sure. Let's go over it on a call when you're back to see if anything sticks out please? In the meantime we've started working based on the @chadwhitacre account that has correct access to unblock ourselves for now.

Finally, looks like @anehzat has the foundations mostly covered. We plan to have a first draft of distributions for your review next week.

**Comment by chadwhitacre** [comment]
> You're a collaborator on `github.com/isaacs/github`

Thanks for digging, I've opened [a new support ticket🔒](https://support.github.com/ticket/personal/0/2341887) with GitHub to be removed as a collaborator from that repo. It is now archived, and I can't find in the UI where to remove myself.

> We've [asked](https://github.com/orgs/community/discussions/67767) in GH community forums.

Gotcha. I've asked my contact at GitHub Sponsors in email to see if we can get an answer here.

> Let's do a quick code review + test to confirm once you're back online please?
> Not really sure. Let's go over it on a call 

Sounds good, let's aim for the first part of next week, will drop to email to schedule. 👍

**Comment by nehzata** [comment]
Awesome! First invoice has been emailed btw.

Hope you're having a great time!

**Comment by nehzata** [comment]
```
curl -s -H 'authorization: bearer <TOKEN>' https://api.github.com/user/installations | jq '.installations | map(.account.login)'
[
  "getsentry-funding",
  "syntaxfm"
]
```

**Comment by chadwhitacre** [comment]
- invoice - cramer to approve, ensure it gets paid promptly
- wkr-gh-sponsor
    - modify to consume CSV from [explore](https://github.com/sponsors/explore?account=getsentry) to ensure 100% deps coverage per GH view
    - validate that it defaults to recurring monthly ("One of our eng noted if Sentry is already billed monthly or are Sponsors invoiced, then using the GraphQL API should create a recurring monthly sponsorship. The sponsorship doesn't need to be specified as monthly as it appears it'll default to monthly. ")
- thanks-dev app install - what perms would getsentry-funding need in order to install for all repos and see them show up in thanks.dev?
  - How do I give @getsentry-funding [explicit permission to access](https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-app-installations-accessible-to-the-user-access-token) the thanks-dev app installations in `getsentry` and `codecov`?

**Comment by chadwhitacre** [comment]
Invoice is approved and should get paid tomorrow. 👍

**Comment by nehzata** [comment]
[wkr-gh-sponsor](https://github.com/thnxdev/wkr-gh-sponsor) is now updated. Doesn't need the csv file. It instead scrapes all the dependencies from GH API.

To run locally (donation disabled):
`. bin/activate-hermit`
`GH_CLASSIC_ACCESS_TOKEN=<TOKEN> ./scripts/wkr-gh-sponsor --config example.config.json`

**Comment by chadwhitacre** [comment]
> these are the only installations currently present for `getsentry-funding` user.
>
> ```
> curl -s -H 'authorization: bearer <TOKEN>' https://api.github.com/user/installations | jq '.installations | map(.account.login)'
> [
>   "getsentry-funding",
>   "syntaxfm"
> ]
> ```

@jeffrey-sentry Any ideas on this one? The [user/installations](https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-app-installations-accessible-to-the-user-access-token) endpoint returns `syntaxfm` for the @getsentry-funding account, but in order to use that account (rather than, e.g., my own account) we need it to also return `getsentry` and `codecov`. Nothing obvious is jumping out at me when I compare:

- https://github.com/orgs/syntaxfm/people/getsentry-funding
- https://github.com/orgs/codecov/people/getsentry-funding
- https://github.com/orgs/getsentry/people/getsentry-funding

Do both `codecov` and `getsentry` use SCIM? Something to do with that, maybe? 🤔

**Comment by chadwhitacre** [comment]
A clue! There is a "Member privileges" settings page, and I find a ["Base permissions" knob](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/setting-base-permissions-for-an-organization) at the top:

<img width="848" alt="Screenshot 2023-09-27 at 10 44 31 AM" src="https://github.com/getsentry/team-ospo/assets/134455/9c942897-686e-48e6-b650-1ba323942c8f">

☝️ That's for `syntaxfm`. For `getsentry` and `codecov` this option is set to "No permission". 😱

<img width="841" alt="Screenshot 2023-09-27 at 10 45 28 AM" src="https://github.com/getsentry/team-ospo/assets/134455/1fb6bba4-d78f-4212-a5b7-0518e385690f">

Does this maybe interact with the user/installations API endpoint? 🤔 I don't want to change the setting permanently, but maybe we could toggle it off temporarily in `syntaxfm` to see if it impacts the endpoint?

**Comment by chadwhitacre** [comment]
@nehzata Let's do this on a call. I've uninstalled the `thanks-dev` app from `syntaxfm`, but it's still showing up in the settings page for @getsentry-funding. My hunch is that you can do something on your end to remove that connection in Thanks.dev, then I can re-add with the base permission turned off in `syntaxfm` to test my hypothesis. That'll probably be most efficient in realtime, so I've sent an invite for later today/tomorrow. In the mean time I've opened [a support request🔒](https://support.github.com/ticket/personal/0/2350182) with GitHub, since if this is true we'll need some workaround since we shouldn't set base perms in `codecov` and `getsentry`.

**Comment by chadwhitacre** [comment]
We confirmed in real time that the hypothesis is correct: we need either "Owner" role on the user account or "Read" base perms for the org in order for the user/installations API to include a given org. We also confirmed that installing the app for only some repos vs. all repos does not make a difference. Assuming GitHub does not get back with a workaround on my support request, our options seem to be:

  1. Use @chadwhitacre (already owner on all three orgs).
  1. Grant @getsentry-funding the owner role.
  1. Change the base permissions to Read in `getsentry` and `codecov`.

Thoughts on how to proceed @jeffrey-sentry @mdtro? Since the [thanks-dev app](https://github.com/apps/thanks-dev/) only has read permissions, iirc using a separate low-privilege account was really a defense-in-depth, yes? I think option (1) may be the least disruptive/risky path forward, assuming no workaround. Can we live with this?

**Comment by chadwhitacre** [comment]
Just saw this come through! 💃

> The payment for invoice **0120** was initiated on **Sep 27, 2023**. We notified **Thanks Dev Pty Ltd** of the payment initiation. The payment should reach the vendor’s bank account in 4-5 business days from the day of this email.

**Comment by nehzata** [comment]
Excellent!

`auto-boost` added to `utils` as well.

**Comment by nehzata** [comment]
Inclusions have been uploaded to both @chadwhitacre & @getsentry-funding accounts

**Comment by chadwhitacre** [comment]
Okay so in my court:

- [ ] get answer on @chadwhitacre account
- [ ] test out `mass-gh-sponsor`
- [ ] test out `auto-boost`

I've pinged our security team internally on the first one.

**Comment by chadwhitacre** [comment]
> if we can differentiate activities from thanks.dev and activities that's performed by you in logs, then I think it's fine to use option 1

**Comment by chadwhitacre** [comment]
Independently, we're in the process of rolling our GH orgs up into a GH [enterprise](https://docs.github.com/en/enterprise-cloud@latest/admin/managing-accounts-and-repositories/managing-organizations-in-your-enterprise)(!). The timing is fortuitous, since it interacts with sponsor accounts. tl;dr I'll have a little dancing to do in their UI before I am ready to configure our new sponsorships.

**Comment by chadwhitacre** [comment]
[Finding this clue](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-with-a-github-app-on-behalf-of-a-user):

> Similarly, if the request triggers a corresponding entry in the audit logs and security logs, the logs will list the user as the actor but will state that the "programmatic_access_type" is "GitHub App user-to-server token".

Looks like we can't filter on `programmatic_access_type` in the audit UI. I've kicked off an export to see if this key shows up in our existing audit log at all.

**Comment by chadwhitacre** [comment]
With help from my Security team, I was able to inspect the audit logs. The `programmatic_access_type` key is present as documented, and additionally there is a "name" field that seems to list the app in question. This gives us what we need to detect any unexpected behavior from the app. Ready to proceed with Thanks.dev under my user.

**Comment by chadwhitacre** [comment]
@nehzata I'm sure you've already addressed this ... Could the Thanks.dev GH App use [installation auth](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation) instead of user auth?

**Comment by mdtro** [comment]
@nehzata , 👋 I'm a part of the Sentry security team. @chadwhitacre Sorry for all the telephone and having you play mediator. 😅

I was reading the ToS to try and get some more detail about how thanks.dev works and is possibly architected. I fully understand you'll need to pull down our code to analyze it, but is it possible to get more clarification as to what is stored long-term on your systems? Questions below.

From the ToS:
> The data we collect from your during the course of providing our Service will be stored according to industry standards. We perform routine backups of our systems and data, however we do not warrant these backups and you should not rely on them for your business continuity.

1. What data is stored? Everything from the repository or just package manifest files?
2. How is the data stored? ie. is the data stored on cloud storage and is the data encrypted?
3. Are backups encrypted and how long are they retained?
4. If the Thanks.dev app is ever deauthorized from the organization, is associated data deleted including backups?

**Comment by nehzata** [comment]
Hi @mdtro ! Thanks for looking into this. Response to questions are below. Please let me know if any other parts need clarification.

> > The data we collect from your during the course of providing our Service will be stored according to industry standards. We perform routine backups of our systems and data, however we do not warrant these backups and you should not rely on them for your business continuity.
> 
> 1. What data is stored? Everything from the repository or just package manifest files?

We only read the manifest files listed [here](https://thanks.dev/static/ecosystems).

> 2. How is the data stored? ie. is the data stored on cloud storage and is the data encrypted?

It is stored in our DB backed by an encrypted EBS in a private EKS cluster. The EKS cluster is configured with role based permissions via IAM roles.

> 3. Are backups encrypted and how long are they retained?

Backups are encrypted on S3. The manifest files + dependency trees are not stored in backups though. We only backup the following DBs:
- users db
- settings db
- accounting db (donations, balances etc)

Everything else will automatically get reanimated in the case of DR. Storage costs would be prohibitive otherwise.

We retain last 30 days via S3 lifecycle policies.

> 4. If the Thanks.dev app is ever deauthorized from the organization, is associated data deleted including backups?

We're not presently deleting any data automatically. In the event of a deauthorization we just detect access has been revoked and stop trying to read the manifest files + updating the dependency tree.

However, we do have the ability to manually handle CCPA and equivalent data deletion requests should there be a need.

**Comment by nehzata** [comment]
> @nehzata I'm sure you've already addressed this ... Could the Thanks.dev GH App use [installation auth](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation) instead of user auth?

@chadwhitacre We don't currently obtain an installation token. Let us look into it and get back to you please.

If I've understood correctly, this would be to differentiate thanks.dev activity from your own activity in the audit logs right?

**Comment by nehzata** [comment]
@chadwhitacre Can we jump on a quick call please to confirm which repos will be included in the dep tree? We'll then be able to do the final export for onboarding. Same time as usual if it's ok with you? 🙏

**Comment by chadwhitacre** [comment]
> If I've understood correctly, this would be to differentiate thanks.dev activity from your own activity in the audit logs right?

Yes, and in general to decouple app activity from my user account.

**Comment by chadwhitacre** [comment]
Ftr looks like we found time to meet today/tonight.

Also, our orgs are now consolidated into an enterprise, ready to proceed with `mass-gh-sponsor`. 👍

**Comment by mdtro** [comment]
> Hi @mdtro ! Thanks for looking into this. Response to questions are below. Please let me know if any other parts need clarification.
> 
> > > The data we collect from your during the course of providing our Service will be stored according to industry standards. We perform routine backups of our systems and data, however we do not warrant these backups and you should not rely on them for your business continuity.
> > 
> > 
> > 
> > 1. What data is stored? Everything from the repository or just package manifest files?
> 
> We only read the manifest files listed [here](https://thanks.dev/static/ecosystems).
> 
> > 2. How is the data stored? ie. is the data stored on cloud storage and is the data encrypted?
> 
> It is stored in our DB backed by an encrypted EBS in a private EKS cluster. The EKS cluster is configured with role based permissions via IAM roles.
> 
> > 3. Are backups encrypted and how long are they retained?
> 
> Backups are encrypted on S3. The manifest files + dependency trees are not stored in backups though. We only backup the following DBs:
> 
> * users db
> * settings db
> * accounting db (donations, balances etc)
> 
> Everything else will automatically get reanimated in the case of DR. Storage costs would be prohibitive otherwise.
> 
> We retain last 30 days via S3 lifecycle policies.
> 
> > 4. If the Thanks.dev app is ever deauthorized from the organization, is associated data deleted including backups?
> 
> We're not presently deleting any data automatically. In the event of a deauthorization we just detect access has been revoked and stop trying to read the manifest files + updating the dependency tree.
> 
> However, we do have the ability to manually handle CCPA and equivalent data deletion requests should there be a need.

@nehzata Thank you for the answers! When you say "you only read manifest files", does that mean that's all that is stored? Or is the entire repo stored, just only those particular files are searched for and read?

**Comment by nehzata** [comment]
@mdtro Apologies for not being clear. We only download and store the manifest files. More specifically we:
- Obtain list of files in repository via GH REST API
- Filter list for manifest files
- Download and store the filtered list of files via GH REST API

**Comment by nehzata** [comment]
> > Can we jump on a quick call please [...] Same time as usual if it's ok with you? 🙏
> 
> Of course, sent for your Thursday AM.
> 
> > to confirm which repos will be included in the dep tree? We'll then be able to do the final export for onboarding.
> 
> My intention is to include all repos for all three orgs (weighted differently, of course). Is this enough to unblock?

Thanks – unblocked!

One more clarifying question – `tag-archived` repositories to be left at default boosting or excluded?

**Comment by chadwhitacre** [comment]
> One more clarifying question – `tag-archived` repositories to be left at default boosting or excluded?

Specific is terrific, I love it! 😁

Here are the topics we're using (per our internal docs):

> - **tag-production**
>     - Definition for production: part of Sentry (including SaaS, ST, any public website, or internal tools we use to keep Sentry running) will be down without this repo
> - **tag-non-production**
>     - Example: your side-projects, hackweek projects, repos that teams use for product management, testing stuff that’s not tied to prod
> - **tag-to-be-archived-{{date-tagged}}**
>     - Repos that we believe can be archived, and will be archived if no action is taken for a period of time.
> - **tag-to-be-production-{{date-tagged}}**
>     - Repos that will be tag as production if no action is taken for a period of time
>     - This will be the default tag for new repos
> - **tag-lost-and-found-{{date-tagged}}**
>     - Repos we don’t know how to classify. Repos with this tag will be moved to `to-be-archived` if no action is taken for a period of time.
> - **tag-archived**

Let's go with:
 - excluded - `tag-archived` `tag-to-be-archived-*` `tag-lost-and-found-*`
 - level 1 - `tag-non-production` `tag-to-be-production-*`
 - level 2 - `tag-production`

Thoughts on two levels of boosting based on repo tags? Doable?

**Comment by chadwhitacre** [comment]
I guess any actually archived repos (regardless of topic) should also be excluded.

I guess any repo without any of the above topics should be ... level 1, let's say? Level 1, let's say.

(I'm half-inclined to treat forks differently than sources, but let's not worry about that this time around.)

**Comment by nehzata** [comment]
Perfect!

Also what % of funds should get allocated to each org?

**Comment by chadwhitacre** [comment]
- ali to investigate 40 (`mass-gh-sponsor`) vs. 77 ([gh](https://github.com/sponsors/explore?account=syntaxfm)) sponsorables for syntaxfm
- chad to get GH to transfer balance from `getsentry` org to `codecov` and `syntaxfm`
- chad to validate that `mass-gh-sponsor` will create recurring donations
- chad to run `mass-gh-sponsor` for all orgs

- ali/armin to recruit to join thanks.dev
- chad to get final security sign-off
- chad to ask GH to confirm that [500+](https://github.com/sponsors/explore?account=getsentry) includes transitive not just first-level deps
- chad to manually boost `codecov` and `syntaxfm` repos
- compare with last year - more inclusions?
- ali to look at adding a setting for "send to GHS/OC even if not on TD"

- firm up the plan for 

- ali/armin to send second invoice by EOW

**Comment by nehzata** [comment]
@chadwhitacre `mass-gh-sponsor` has been updated to support `import-csv` command. [doc](https://github.com/thnxdev/utils/blob/master/README.md#22-run-locally-import-from-csv)

**Comment by nehzata** [comment]
@chadwhitacre Is it ok if we revert the max depth settings back to 1 please?

**Comment by chadwhitacre** [comment]
I need to filter out Sentry employees.

**Comment by chadwhitacre** [comment]
```
#!/usr/bin/env zsh

for i in {1..10}; do
  curl "https://github.com/enterprises/sentry/people?page=$i" \
    -H 'authority: github.com' \
    -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
    -H 'accept-language: en-US,en;q=0.9' \
    -H 'cache-control: max-age=0' \
    -H "cookie: $COOKIE" \
    -H 'sec-ch-ua: "Google Chrome";v="117", "Not;A=Brand";v="8", "Chromium";v="117"' \
    -H 'sec-ch-ua-mobile: ?0' \
    -H 'sec-ch-ua-platform: "macOS"' \
    -H 'sec-fetch-dest: document' \
    -H 'sec-fetch-mode: navigate' \
    -H 'sec-fetch-site: same-origin' \
    -H 'sec-fetch-user: ?1' \
    -H 'upgrade-insecure-requests: 1' \
    -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36' \
    --compressed \
    > "$i.html"
done
```

```
$ grep 'id="user-' *.html | sed -e 's/^.*id="user-\([^"]*\)".*$/\1/' | sort | uniq > members
$ wc -l members 
     279 members
```

👍

**Comment by chadwhitacre** [comment]
Blech. Realizing that a fair number of people use separate user accounts for personal and work, so e.g. `asottile` shouldn't be funded but `asottile-sentry` is the one we can easily see programmatically.

**Comment by chadwhitacre** [comment]
```python
#!/usr/bin/env python3
import csv
import os

members = set(open('members').read().splitlines())

for filename in os.listdir('.'):
    if not filename.startswith('github-explore-sponsors-for-'):
        continue
    print(f'processing {filename}')
    r = csv.reader(open(filename))
    w = csv.writer(open('filtered-'+filename, 'w+'))

    w.writerow(next(r)) # headers

    for row in r:
        if row[0] in members:
            print(f'skipping {row[0]}')
            continue
        w.writerow(row)
```

After manually adding `asottile` to `members`:

```
$ ./filter-out-members.py 
processing github-explore-sponsors-for-syntaxfm-2023-10-04.csv
processing github-explore-sponsors-for-getsentry-2023-10-04.csv
skipping mitsuhiko
skipping asottile
processing github-explore-sponsors-for-codecov-2023-10-04.csv
skipping asottile
```

```
sqlite> delete from donations where recipient_id='mitsuhiko';
sqlite> delete from donations where recipient_id='asottile';
sqlite> select sponsor_id, count(*) n from donations group by sponsor_id order by n desc;
getsentry|535
codecov|311
syntaxfm|77
sqlite>
```

I did a quick skim of `select distinct(recipient_id) from donations order by recipient_id;`, nobody jumping out at me but it was a quick skim.

**Comment by chadwhitacre** [comment]
Pretty close to running this, I want to:

- double-check that $4/ea is the right amount
- sort out what to do with yyx990803, we have a $500/mo set up still/again, there was something with that coming from marketing budget last time
- think about dupes, account is in 2 or 3 orgs, how do I set it up

**Comment by chadwhitacre** [comment]
Testing with one.

```
sqlite> delete from donations where recipient_id != 'sindresorhus';
sqlite> select * from donations;
927|getsentry|sindresorhus|1696442229|0|0
sqlite>
```

**Comment by chadwhitacre** [comment]
```
utils🐚 $ mass-gh-sponsor donate --gh-classic-access-token=$TOKEN --amount=4
INFO[0000] starting                                     
INFO[0000] donating getsentry:sindresorhus              
utils🐚 $
```

💃 

<img width="658" alt="Screenshot 2023-10-04 at 1 58 53 PM" src="https://github.com/getsentry/team-ospo/assets/134455/b47f635d-04d0-4ca7-9996-dc301946055c">

**Comment by chadwhitacre** [comment]
I did a one-time as a test.

<img width="828" alt="Screenshot 2023-10-04 at 4 22 14 PM" src="https://github.com/getsentry/team-ospo/assets/134455/fe7fc64d-f2b9-41d0-b093-454be829d000">

**Comment by chadwhitacre** [comment]
I did a one-time as a test.

<img width="648" alt="Screenshot 2023-10-04 at 4 41 36 PM" src="https://github.com/getsentry/team-ospo/assets/134455/39e062a7-7590-4b13-9e1f-c5dcf68fca22">

**Comment by chadwhitacre** [comment]
Seeing a lot of minimums.

```
ERRO[0116] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $9"
ERRO[0151] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0154] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0154] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0157] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0158] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0161] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $8"
ERRO[0161] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0162] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $11"
ERRO[0164] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0164] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $8"
ERRO[0167] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0168] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0169] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $10"
ERRO[0170] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $10"
ERRO[0171] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0171] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0171] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $5"
ERRO[0172] failed to create sponsorship                  error="Could not create new Sponsors tier: Monthly price must be at least $500"
```

I captured some but not all of the logs. Looks like I can use timestamp columns to determine which errored.

```
sqlite> select * from donations where recipient_id='emberjs';
1468|getsentry|emberjs|1696453366|0|1696453817
sqlite> select * from donations where recipient_id='wooorm';
1470|getsentry|wooorm|1696453366|1696453819|0
sqlite>
```

Will pick up there later.

**Comment by chadwhitacre** [comment]
Not going to reach 100 but should be able to get to like 97%?

<img width="243" alt="Screenshot 2023-10-04 at 5 46 30 PM" src="https://github.com/getsentry/team-ospo/assets/134455/75a802e0-ff15-4712-b94a-0ed671da0754">

**Comment by nehzata** [comment]
> Seeing a lot of minimums.

What if it's not recurring? Do minimum's still apply?

**Comment by nehzata** [comment]
The following query should return the errored donations:

`SELECT sponsor_id, recipient_id FROM donations WHERE donate_attempt_ts > 0;`

**Comment by nehzata** [comment]
`UPDATE donations SET donate_attempt_ts = 0 where id in (SELECT id FROM donations WHERE donate_attempt_ts > 0 LIMIT 1) RETURNING *;`

Will reset one of the failed donations so it can get retried. Will return the corresponding row.

**Comment by chadwhitacre** [comment]
@nehzata I just shared some feedback in private email re: recruiting emails. Core concern:

> when I connect my github account they want access to all my public and private repos. It looks legit, but it seems sketchy.

I know we've discussed this before. Is there anything we can do here for maintainers vs. sponsors?

Secondarily (and it may be too late for this), is it worth cc'ing me on the the emails or other outreach (twitter) to increase legitimacy?

**Comment by nehzata** [comment]
> I know we've discussed this before. Is there anything we can do here for maintainers vs. sponsors?

They have the option of only selecting one of their public repositories instead of all. We'll add this in more detail to the FAQs.

> Secondarily (and it may be too late for this), is it worth cc'ing me on the the emails or other outreach (twitter) to increase legitimacy?

That would be great if you don't mind please. We've only sent 27 messages so far. 

We're also adding a link to the FAQs page in the email. 🤞

**Comment by chadwhitacre** [comment]
Okay I am rerunning `mass-gh-sponsors donate` with better log management. Create fails if there is an existing active sponsorship, so all the ones I created yesterday should be fine, and I can get a definitive list of the minimums in our list.

**Comment by chadwhitacre** [comment]
```
utils🐚 $ cat log|grep error|wc -l
     534
utils🐚 $
```

👍

**Comment by chadwhitacre** [comment]
Ftr we've got a thumbs-up from @mdtro on https://github.com/getsentry/team-ospo/issues/108#issuecomment-1745531048, and I've confirmed in private Slack that we're all cleared security-wise to proceed with the thanks-dev app using my account. 👍

**Comment by nehzata** [comment]
@chadwhitacre a few updates:
- We've implemented the first stage of non TD funding path. Dependencies page shows all accounts seeking funds (logged into TD + GH Sponsors + OC).
- Funds have hit our bank account.
- Second invoice sent.
- Plan to pay foundation invoices by EOW.

**Comment by nehzata** [comment]
@chadwhitacre The second spreadsheet has been updated to include the calculations. Let's copy paste into master spreadsheet once boost factors are configured as there are email addresses in this one?

**Comment by nehzata** [comment]
The following foundations have been paid as of today:
- Rust
- Apache
- PHP
- Ruby
- Outreachy

Awaiting response from:
- Python
- Javascript
- PostgreSQL

**Comment by chadwhitacre** [comment]
One-time is not a workaround (at least through the UI, not sure about API but seems unlikely to work). :-/

<img width="351" alt="Screenshot 2023-10-06 at 9 53 48 AM" src="https://github.com/getsentry/team-ospo/assets/134455/b7e1e3f8-e73b-4205-b2e7-cc69368a2e00">

**Comment by chadwhitacre** [comment]
<img width="229" alt="getsentry" src="https://github.com/getsentry/team-ospo/assets/134455/bcd8b063-4b02-4fda-b284-fca569b3470d">

https://twitter.com/chadwhitacre_/status/1710378308012023998

**Comment by chadwhitacre** [comment]
<img width="587" alt="Screenshot 2023-10-06 at 8 37 17 PM" src="https://github.com/getsentry/team-ospo/assets/134455/b2bddd3e-d322-4190-8a20-052c98ab975c">

https://twitter.com/chadwhitacre_/status/1710399485073846493 😬

**Comment by chadwhitacre** [comment]
<img width="594" alt="Screenshot 2023-10-10 at 10 16 26 AM" src="https://github.com/getsentry/team-ospo/assets/134455/15e58bb4-981a-4334-8808-5fb12c401afa">

https://twitter.com/zeeg/status/1710764429519049192

**Comment by nehzata** [comment]
@chadwhitacre The [internal working spreadsheet](https://docs.google.com/spreadsheets/d/1ek_Tz1nnGR1pSkTVasDtPcN5eaVP-vcJQlzH26rM8AA/edit?usp=sharing) has the boost factors added to it in column G. I'm still working on merging last year's numbers into it. Once done will copy into master spreadsheet.

**Comment by chadwhitacre** [comment]
Asks on call (order of priority):

1. implement a $5/mo minimum (no 17¢ per month)
2. drop cents entirely, whole dollar amounts only
3. put maintainers into ≲ 10 buckets (e.g. 5 10 20 50 100 200 500 1000 2000/mo)
    - easier to reason about and compare one to another
5. iterate on boosting concept at each level
    1. **org** - maybe just include / exclude? - less import orgs have fewer repos, so weighting takes care of itself
    2. **repo** - prod / non-prod / exclude
    3. **maintainer** - put into buckets with whole dollar adjustment per maintainer
    4. **ecosystem** - would like to be able to see this as a percentage of total dollar volume

It's okay to treat the $24,291/mo (per below, from [sheet](https://docs.google.com/spreadsheets/d/1efxcQPg_4bG-zWh59m_-1Bz7cawLD36oJcLEPV9-Sa8/edit#gid=849463399)) as a max, i.e., given the above constraints, get as close as possible without going over.

<img width="314" alt="Screenshot 2023-10-12 at 7 33 56 PM" src="https://github.com/getsentry/team-ospo/assets/134455/c8b30ef9-4b80-4c65-9ddf-379f9c50ff3d">

I realize this is a lot to bring up at this point. Let's see how far we can get.

**Comment by nehzata** [comment]
@chadwhitacre Quick update – apologies I haven't had a chance to update the spreadsheet yet. Still working through the onboarding. We've finished first phase of outreach but the backlog of dependency analysis is still going. Combination of GH rate limiting, npm updates + ~20k new repos have logged in. Optimising the process as a fair number of users are seeing "collation not available" atm.

I think I have a solution to the bucketing requirement. Plan to test it tomorrow 🤞 . Will post another update COB.

**Comment by chadwhitacre** [comment]
Thanks for the update @nehzata. My priority would be to send money this week one way or another. Let's tune our efforts to that. The good news is that since we're running monthly we can improve the system in the coming months.

Good problem to have, onboarding so many new maintainers! 😁

**Comment by nehzata** [comment]
@chadwhitacre We've figured out a way to implement minimums. Implementing today. Sample of it is in `minimums` tab in the spreadsheet. What do you think?

**Comment by nehzata** [comment]
@chadwhitacre Minimums have been implemented. How are you looking for a quick catch up to confirm the numbers please?

**Comment by chadwhitacre** [comment]
Awesome! Just sent an invite, hopefully we can jump on a call?

**Comment by nehzata** [comment]
Awesome! Thanks!

**Comment by nehzata** [comment]
Minimums deployed (currently set to $10/mo) + spreadsheet updated (2023 tab)!

**Comment by chadwhitacre** [comment]
Awesome! I'll check it out ...

**Comment by chadwhitacre** [comment]
1. ali/armin to review 77 not founds and add to inclusions as appropriate
2. chad to review sheet and set weight/boost as desired - Friday afternoon (SF)
3. ali/armin to import them in to TD - Saturday (Sydney)
4. chad to do final review within TD - Sunday evening (Pittsburgh)
5. rock and roll! 🎸 - Monday (Sydney)

**Comment by nehzata** [comment]
@chadwhitacre Can you please review the "NOT FOUND" list (now reduced to 68)? I've marked the ones that don't need to be imported as they've moved or are detected at org level. The remainder seem to be from last year's survey results?

**Comment by chadwhitacre** [comment]
> Can you please review the "NOT FOUND" list (now reduced to 48)?

Ship it! :shipit:

**Comment by chadwhitacre** [comment]
Any chance we could make https://thanks.dev/d/gh/getsentry/dependencies not indicative? 🐭

**Comment by chadwhitacre** [comment]
Okay! @nehzata and I just finalized Step 2 of https://github.com/getsentry/team-ospo/issues/108#issuecomment-1771858654 ... a little behind schedule but we're hoping to catch up. Ali is importing now, I will check back in a bit to do a final review, and then we will be cleared to run our first payments! 🤞

**Comment by chadwhitacre** [comment]
I've just done a final review in TD, there are one or two discrepancies that @nehzata has walked me through, I've given him the go ahead to adjust in a way that makes sense and to proceed with making our first payouts! 👍

**Comment by chadwhitacre** [comment]
Caught a couple issues:

- Excluded a few more (`swatinem` is an employee, `pre-commit` and `pycqa` are `asottile`'s).
- We need to includes Fastlane, Moka, and RRWeb (from the ticket description). Might need to use Python bonus for this?

**Comment by chadwhitacre** [comment]
Worked it through on a call and we've got something we can ship. There's always next month if something is egregiously out of whack.

**Comment by nehzata** [comment]
@chadwhitacre Confirming first round has been distributed. Emails to recipients already logged into TD will start going out later today. Armin & I will reach out and onboard the other recipients as discussed. 🎉

**Comment by chadwhitacre** [comment]
Woo-hoo! 😍 Thank you guys for everything! Could not have done it without you this year! 🙏

**Comment by nehzata** [comment]
Public donor pages are now working as expected:
https://thanks.dev/d/gh/getsentry/dependencies
https://thanks.dev/d/gh/getsentry/dependencies

**Comment by chadwhitacre** [comment]
Sweet! I'll look at updating the blog post, it's scheduled for 9:30 SF time.

**Comment by chadwhitacre** [comment]
I'm surprised there are only 20 recipients on Codecov, and that some are below the $10 minimum.

**Comment by nehzata** [comment]
Codecov fixed! Sorry about that!

**Comment by chadwhitacre** [comment]
Okay phew. 😅

**Comment by nehzata** [comment]
Thanks for the shout out in the blog post btw! Has been a blast working with you Chad!

**Comment by chadwhitacre** [comment]
@anehzat Can you link in a comment where our logo is placed? Understood that for Postgres we don't get placement.

**Comment by chadwhitacre** [comment]
[Some feedback on HN](https://news.ycombinator.com/item?id=38013761):

> I missed the part about this being a recurring monthly payout - $180 does clear the threshold of what I'd consider a significant donation, so I appreciate that.
> I don't mind it being spread out monthly. I think it would help if the process made it a bit more clear what to expect (in terms of the total payout and timeline) in the initial message and/or the UI.

**Comment by chadwhitacre** [comment]
Calling this shipped!

**Comment by nehzatarmin** [comment]
> @anehzat Can you link in a comment where our logo is placed? Understood that for Postgres we don't get placement.

@chadwhitacre Here's the link for the ones that are currently have the logo 
https://opencollective.com/phpfoundation
<img width="1128" alt="image" src="https://github.com/getsentry/team-ospo/assets/120353582/c0fd662c-b057-4a1c-96c7-864c326ee568">

https://apache.org/foundation/sponsors 
<img width="1385" alt="image" src="https://github.com/getsentry/team-ospo/assets/120353582/b5ce1308-37cb-43b8-b617-74fd1d77ebc0">

https://foundation.rust-lang.org/members/
<img width="1582" alt="image" src="https://github.com/getsentry/team-ospo/assets/120353582/e5f015b3-db89-4b5f-aae6-8f7964fef7df">

https://openjsf.org/
<img width="1341" alt="image" src="https://github.com/getsentry/team-ospo/assets/120353582/1a60f758-5ec5-4711-9d7d-4ec7d16570c2">

The ones I'm following up with still are Outreachy, Ruby, Postgres & Python.
Let me know who's your contact at OSI & I can also reach out to them.

**Comment by chadwhitacre** [comment]
> Let me know who's your contact at OSI & I can also reach out to them.

Sent to you in email. Thanks!

**Comment by nehzatarmin** [comment]
nice OSI is also marked as done :) 
<img width="1521" alt="image" src="https://github.com/getsentry/team-ospo/assets/120353582/4736331f-323f-489d-aa1d-66f7044e4add">

**Comment by chadwhitacre** [comment]
@nehzatarmin Last items I'm seeing on the TODO here are:

- pay Python - email traffic on this, is payment complete?
- logo on Python
- logo on Outreachy

**Comment by anehzat** [comment]
thanks @chadwhitacre 

- We just sent the payment receipt & follow up to PFS
- We sent another followup to Outreachy

I've put a reminder for myself to follow up biweekly with them about it.

**Comment by chadwhitacre** [comment]
Found us on Outreachy! 🙌 

<img width="1064" alt="Screenshot 2024-03-01 at 1 22 24 PM" src="https://github.com/getsentry/team-ospo/assets/134455/be5c8e1e-71e8-4450-9c8e-1549db33b868">
